src/Avalonia.Animation/Keyframes/KeyFrame.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.ComponentModel;
+
+namespace Avalonia.Animation.Keyframes
+{
+
+    /// <summary>
+    /// Stores data regarding a specific key
+    /// point and value in an animation.
+    /// </summary>
+    public class KeyFrame
+    {
+        internal bool timeSpanSet, cueSet;
+        private TimeSpan _ktimeSpan;
+        private Cue _kCue;
+
+        /// <summary>
+        /// Gets or sets the key time of this <see cref="KeyFrame"/>.
+        /// </summary>
+        /// <value>The key time.</value>
+        public TimeSpan KeyTime
+        {
+            get
+            {
+                return _ktimeSpan;
+            }
+            set
+            {
+                if (cueSet)
+                {
+                    throw new InvalidOperationException($"You can only set either {nameof(KeyTime)} or {nameof(Cue)}.");
+                }
+                timeSpanSet = true;
+                _ktimeSpan = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the cue of this <see cref="KeyFrame"/>.
+        /// </summary>
+        /// <value>The cue.</value>
+        public Cue Cue
```

Comment:
So a `Cue` is a time as a percent of the animation and a `KeyTime` is an actual time?

Reply:
Yes, i added the `KeyTime` as an alternative when they want to exactly pinpoint when in the animation iteration a keyframe should be activated and also they can't set both `Cue` and `KeyTime` at the same time :D